packages/components/src/templates/next/components/complex/Video/Video.tsx

```diff
@@ -0,0 +1,105 @@
+import type { VideoProps } from "~/interfaces"
+import { VIDEO_EMBED_URL_PATTERN } from "~/utils/validation"
+import { ComponentContent } from "../../internal/customCssClass"
+
+const VALID_VIDEO_DOMAINS = {
+  youtube: [
+    "www.youtube.com",
+    "www.youtube-nocookie.com",
+    "youtube.com",
+    "youtube-nocookie.com",
+  ],
+  vimeo: ["player.vimeo.com"],
+  fbvideo: ["www.facebook.com"],
+}
+
+const isValidVideoUrl = (url: string) => {
+  if (!url) {
+    return false
+  }
+
+  try {
+    const urlObject = new URL(url)
+    const allValidVideoDomains = Object.values(VALID_VIDEO_DOMAINS).flat()
+
+    return (
+      allValidVideoDomains.includes(urlObject.hostname) &&
+      new RegExp(VIDEO_EMBED_URL_PATTERN).test(url)
+    )
```

Comment:
is there a need for the first 2 checks if we are aldy gonna do a regexp against `VIDEO_EMBED_URL_PATTERN`? seems like double effort + duplicating of logic in 2 sources

Reply:
Same reasoning as for Maps.